contracts/farms/common/RevoFarmBot.sol

```diff
@@ -3,25 +3,38 @@ pragma solidity 0.8.4;
 
 import "hardhat/console.sol";
 
-import "./IMoolaStakingRewards.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Router02.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUbeswapRouter.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Pair.sol";
-import "./IRevoFees.sol";
-import "./openzeppelin-solidity/contracts/ERC20.sol";
-import "./openzeppelin-solidity/contracts/AccessControl.sol";
-import "./openzeppelin-solidity/contracts/SafeERC20.sol";
-import "./openzeppelin-solidity/contracts/Pausable.sol";
-
-contract UbeswapFarmBot is ERC20, AccessControl, Pausable {
+import "../../openzeppelin-solidity/contracts/ERC20.sol";
+import "../../openzeppelin-solidity/contracts/AccessControl.sol";
+import "../../openzeppelin-solidity/contracts/SafeERC20.sol";
+import "../../openzeppelin-solidity/contracts/Pausable.sol";
+import "../../fees/interfaces/IRevoFees.sol";
+
+/**
+ * RevoFarmBot is an abstract base class that can be used to implement Revo farm bots on top
+ * of arbitrary sources of yield. These sources of yield come from staking some ERC20 token
+ * (the "staking token") into a contract.
+ *
+ * Minimally, subclasses must implement the _deposit, _withdraw, and _claimRewards methods that
+ * deposit/withdraw an amount of the staked token into the "farm", and claim rewards. This will
+ * result in a RevoFarmBot that does not auto-compound yield; it acts only as "storage" for a
+ * user's staking token.
+ *
+ * Notably, this class does not contain a virtual "compound" function; this is because depending
+ * on the implementation, a function repsonsible for compounding the farm bot's staked position may
+ * require different arguments from the caller.
+ *
+ * This class also does not make any assumptions about how yield rewards are converted back into
+ * the staking token. For example, this may require a series of swaps and minting a liquidity token ("LP"),
+ * or for a single-staking source of yield that issues the staking token itself as a reward, nothing at all.
```

Comment:
is there an example you have in mind of a "single-staking source of yield that issues the staking token itself as a reward"? Specifically one where a farm bot would be useful?